src/prog_models/linear_model.py

```diff
@@ -67,35 +104,66 @@ def _propertyCheck(self, rowsCount : int, colsCount : int, notes : list) -> None
         matrix = getattr(self, notes[0]) 
         if not isinstance(matrix, np.ndarray):
             raise TypeError("Matrix type check failed: @property {} dimensions is not of type list or NumPy array.".format(notes[0]))
+        
 
         matrixShape = matrix.shape
-        if (matrixShape[0] != rowsCount or # check matrix is 2 dimensional, correspond to rows count
-            len(matrixShape) == 1 or # check .shape returns 2-tuple, meaning all rows are of equal length
-            matrixShape[1] != colsCount or # check all rows are equal to correct column count
-            matrix.ndim != 2): # check matrix is 2 dimensional
-            raise AttributeError("Matrix size check failed: @property {} dimensions improperly formed along {} x {}.".format(notes[0],notes[1],notes[2]))
-   
+
+        if (matrix.ndim != 2 or # Checks too see if matrix is two-dimensional
+            matrixShape[0] != rowsCount or # checks if matrix has correct row count 
+            matrixShape[1] != colsCount): # check all rows are equal to correct column count
+                raise AttributeError("Matrix size check failed: @property {} dimensions improperly formed along {} x {}.".format(notes[0],notes[1],notes[2]))
+
     @property
-    @abstractmethod
     def A(self):
-        pass
+        return self.parameters['_A']
+
+    @A.setter
+    def A(self, value):
+        self.parameters['_A'] = value
 
     @property
     def B(self):
-        return np.zeros((self.n_states, self.n_inputs))
+        return self.parameters['_B']
+
+    @B.setter
+    def B(self, value):
+        if (value is None):
+            self.parameters['_B'] = np.zeros((self.n_states, self.n_inputs))
+        else:
+            self.parameters['_B'] = value
 
     @property
-    @abstractmethod
     def C(self):
-        pass
+        return self.parameters['_C']
+
+    @C.setter
+    def C(self, value):
+        if (value is None):
+            self.parameters['_C'] = np.zeros((self.n_outputs, self.n_states))
+        else:
+            self.parameters['_C'] = value
```

Comment:
Since we're adding setters here, let's do a check before overriding it. 

Something like this:

```suggestion
            old_value = self.parameters['_C']
            self.parameters['_C'] = value
            try:
                 self._propertyCheck(self.n_outputs, self.n_states, ["C","outputs","states"])
            except (TypeError, AttributeError) as ex:
                 # Unacceptable value, reset and re-raise
                 self.parameters['_C'] = old_value
                 raise ex
```

What do you think?